dipy/meson.build

```diff
@@ -269,6 +269,17 @@ py3.install_sources(
   subdir: 'dipy'
 )
 
+# ------------------------------------------------------------------------
+# Manage datafiles
+# ------------------------------------------------------------------------
+
+data_install_dir = join_paths(get_option('datadir'), 'share', 'doc', meson.project_name())
```

Comment:
datadir is already share so you should skip repeating it.

Reply:
This is what I believed too. However, in my conda virtual environment, it does not return the 'share' folder. 

There is something strange... let me try again

Reply:
You can run `meson configure` in the build directory and it will print out all builtin/project options along with their current values.

Reply:
ok, I checked again, still does not work, not sure what is wrong:

- I can see `datadir = share` with `meson configure``
- after using `pip install dipy`, the `share` folder is empty. 
`ls: /opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/share/doc/dipy/examples: No such file or directory`
-  however, I got all my examples in `doc/dipy/examples`.

```
❯ ls /opt/homebrew/Caskroom/miniforge/base/envs/dipy-39-x86/doc/dipy/examples
__pycache__                               reconst_csa.py                            segment_clustering_metrics.py
affine_registration_3d.py                 reconst_csa_parallel.py                   segment_extending_clustering_framework.py
affine_registration_masks.py              reconst_csd.py                            segment_quickbundles.py
afq_tract_profiles.py                     reconst_csd_parallel.py                   simulate_dki.py
brain_extraction_dwi.py                   reconst_cti.py                            simulate_multi_tensor.py
bundle_assignment_maps.py                 reconst_dki.py                            snr_in_cc.py
bundle_extraction.py                      reconst_dki_micro.py                      streamline_formats.py
bundle_group_registration.py              reconst_dsi.py                            streamline_length.py
bundle_registration.py                    reconst_dsi_metrics.py                    streamline_registration.py
bundle_shape_similarity.py                reconst_dsid.py                           streamline_tools.py
bundlewarp_registration.py                reconst_dti.py                            surface_seed.py
cluster_confidence.py                     reconst_forecast.py                       syn_registration_2d.py
combined_workflow_creation.py             reconst_fwdti.py                          syn_registration_3d.py
contextual_enhancement.py                 reconst_gqi.py                            tissue_classification.py
denoise_ascm.py                           reconst_ivim.py                           tracking_bootstrap_peaks.py
denoise_gibbs.py                          reconst_mapmri.py                         tracking_deterministic.py
denoise_localpca.py                       reconst_mcsd.py                           tracking_introduction_eudx.py
denoise_mppca.py                          reconst_msdki.py                          tracking_pft.py
denoise_nlmeans.py                        reconst_qtdmri.py                         tracking_probabilistic.py
denoise_patch2self.py                     reconst_qti.py                            tracking_ptt.py
fast_streamline_search.py                 reconst_qtiplus.py                        tracking_rumba.py
fiber_to_bundle_coherence.py              reconst_rumba.py                          tracking_sfm.py
gradients_spheres.py                      reconst_sfm.py                            tracking_stopping_criterion.py
histology_resdnn.py                       reconst_sh.py                             viz_advanced.py
kfold_xval.py                             reconst_shore.py                          viz_bundles.py
linear_fascicle_evaluation.py             reconst_shore_metrics.py                  viz_roi_contour.py
motion_correction.py                      register_binary_fuzzy.py                  viz_slice.py
path_length_map.py                        reslice_datasets.py                       workflow_creation.py
piesno.py                                 restore_dti.py
quick_start.py                            segment_clustering_features.py
```

Reply:
it looks like `get_option('datadir')` return an empty despite `meson configure` giving me the correct default value

Reply:
```
❯ meson --version
1.2.1
```

Reply:
This appears to be a meson-python bug. It maps "datadir" to /usr instead of /usr/share -- or more generally, to the python prefix instead of python prefix + "/share".

Can you report it to the meson-python bug tracker?

Note that this would work with `spin build`.

Reply:
> This appears to be a meson-python bug. It maps "datadir" to /usr instead of /usr/share -- or more generally, to the python prefix instead of python prefix + "/share".

Thank you for your help and finding that

> Can you report it to the meson-python bug tracker?

ok

> Note that this would work with spin build.

I confirm

Reply:
Hi @eli-schwartz, 

What would you recommend as an alternative or "best practices" concerning data files? 

Not sure I want to keep 2 different behaviors between spin and meson-python. I do not think those datafiles should be inside the DIPY package.

Thank you for your feedback.

Reply:
I think that it should be relatively simple to fix in meson-python as a long term solution.

One possible "quick hack" is to add a meson_options.txt option called "meson-python" and pass it as a setup arg in the meson-python section of the pyproject.toml -- default it to false and have pip install (via meson-python) set it to true.

Spin will use meson directly and install to `get_option('datadir')`. If the "meson-python" option is truthy then install to `get_option('datadir') / 'share'` with a comment "workaround for meson-python bug: LINK".

Like I said, it's a hack. :D